test/lit/isorecursive-whole-group.wast

```diff
@@ -0,0 +1,21 @@
+;; TODO: Autogenerate these checks! The current script cannot handle `rec`.
+
+;; RUN: wasm-opt %s -all --hybrid -S -o - | filecheck %s
+
+;; Check that unused types are still included in the output when they are part
+;; of a recursion group with used types.
+
+(module
+
+;; CHECK:      (rec
+;; CHECK-NEXT:  (type $used (struct_subtype data))
+;; CHECK-NEXT:  (type $unused (struct_subtype data))
+;; CHECK-NEXT: )
+
+ (rec
+  (type $used (struct_subtype data))
+  (type $unused (struct_subtype data))
+ )
```

Comment:
Please add a test of a rec group of size 1, which gets different code paths in this PR. Also of size 0, if that is valid?

Reply:
Sounds good, will add a test for a group of size 1. Groups of size zero cannot exist, since the only way to get a rec group is to query a `HeapType` that must be a member of the returned group.